src/main/java/org/openmrs/addonindex/domain/artifactory/VersionList.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.addonindex.domain.artifactory;
+
+import org.openmrs.addonindex.util.Version;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+public class VersionList {
+
+    private SortedSet<Version> versions;
+
+    private static final String[] STRINGS_TO_EXCLUDE = {"alpha", "beta", "hibernate", "maven", "RC", "SNAPSHOT"};
+
+    public VersionList(Collection<String> coreversions) {
+        this.versions = new TreeSet<>();
+        for (String candidateVersion : coreversions) {
+            if (!stringContainsItemFromList(candidateVersion, STRINGS_TO_EXCLUDE)) {
+                this.versions.add(new Version(candidateVersion.replaceAll("/", "")));
```

Comment:
The only method that calls this constructor is the VersionsService which was the one initially responsible for cleaning the string. If I am not mistaken , you had asked me to move it to versionlist , the reason being that every time the rest api would call a function, the string would be screened all over again which is not needed. Hence it's now modified such that it is only screened when the Scheduler class sets the versions